### PR TITLE
APEXCORE-310 - Apex cli support to kill the app by appname

### PR DESCRIPTION
**Pull request by deepak-narkhede**
This changes adds support to kill the application by application name for apex cli.

Unit Testing Completed: Please find the attached tested scenarios.
[testing-kill-support.txt](https://github.com/apache/apex-core/files/499819/testing-kill-support.txt)

### CODE REVIEW

**Comment by deepak-narkhede** [comment]
@tushargosavi and @yogidevendra Can you please review the request.

**Comment by sandeshh** [comment]
Yarn doesn't support uniqueness for App Names, but Apex implements uniqueness for names. So the assumption in the implementation should be clearly spelled out.

File: engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java [hunk 1]
```diff
@@ -962,6 +962,24 @@ private static String expandCommaSeparatedFiles(String filenames) throws IOExcep
     return result.toString();
   }
 
+  protected ApplicationReport getApplicationByName(String appName)
+  {
+    List<ApplicationReport> appList = getApplicationList();
+    if (appName == null) {
```
Comment (tushargosavi) [inline engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java:966]: move this check at start of the function.

**Comment by tushargosavi** [comment]
The pull request looks fine to me. If there are no objections then I will merge it

File: engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java (hunk 1 above)
Comment (deepak-narkhede) [reply engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java:966]: Done. Thanks !

**Pull request merged**